flake.nix

```diff
@@ -29,16 +29,27 @@
     {
       nixosModule = self.nixosModules.nixos-vault-service;
       nixosModules = {
-        nixos-vault-service = import ./module/implementation.nix;
+        nixos-vault-service = {
+          imports = [
+            ./module/implementation.nix
+          ];
+
+          nixpkgs.overlays = [
+            self.overlays.default
+          ];
+        };
       };
 
       packages = forAllSystems
-        ({ pkgs, ... }: {
+        ({ pkgs, ... }: rec {
           messenger = pkgs.callPackage ./messenger { };
+
+          default = messenger;
         });
 
-      defaultPackage = forAllSystems
-        ({ system, ... }: self.packages.${system}.messenger);
+      overlays.default = final: prev: {
+        messenger = self.packages.${final.stdenv.system}.messenger;
```

Comment:
I wonder if we should namespace this (e.g. `detsys-messenger`), so that if a future Nixpkgs revision introduces a package by the same name, we won't be overwriting it?

Reply:
Yes, probably a good idea.